### ISSUE TITLE
Name for choice value incorrectly truncated

### ISSUE THREAD

**Issue by jhendrixMSFT** [Azure/autorest.modelerfour#275]
**Describe the issue**
In compute 2019-12-01 HardwareProfile defines a large (unsealed) choice.  One of the choices has the name truncated resulting in duplicate names.

```
"Standard_GS3",
"Standard_GS4",
"Standard_GS5",
"Standard_GS4-8",
"Standard_GS4-4",
"Standard_GS5-16",
```

Choice `Standard_GS4-4` is incorrectly named `StandardGS4` which already exists.  It should be named `StandardGS44`.

** REQUIRED ** 
**ModelerFour** version: 4.13.339
**AutoRest-core** version: 3.0.6274
**CommandLine** :
OpenAPI/Swagger file:
https://github.com/Azure/azure-rest-api-specs/blob/master/specification/compute/resource-manager/Microsoft.Compute/stable/2019-12-01/compute.json

**Comment by pakrym** [comment]
Blocking one of our target libraries.

**Comment by daviwil** [comment]
It looks like this is due to some logic in the style formatters we use for names which removes one of the pair adjacent duplicate entries in the name (`4` and `4` in this case).  There's a way to disable this behavior conditionally for different name types but I want to check with @fearthecowboy first to determine whether it's "safe" to turn it off for enums.

**Comment by fearthecowboy** [comment]
I'd be nervous of arbitrarily turning it off for a class of things, but I don't see another way. 

Enum value names shouldn't ever be accidentally including duplicate words (unless they used duplicate words in the value itself?) , so ... yeah, turning it off for enum values may not be a bad idea at all.

**Comment by daviwil** [comment]
This is fixed in the modelerfour version [4.13.348](https://github.com/Azure/autorest.modelerfour/releases/tag/v4.13.348).

**Comment by daviwil** [comment]
Confirmed fixed by @jhendrixMSFT, closing.
